### ISSUE TITLE
Multiple service instances with Hystrix and Turbine AMQP

### ISSUE THREAD

**Issue by aminmc** [spring-cloud/spring-cloud-netflix#472]
Hello

I have two instances of a micro service with the following HsytrixCommand:

```java
@HystrixCommand(commandKey = "PersistDocumentCommand", groupKey = "DocumentManagementApp")
    public void persist(Document document) {
        hazelcastInstance.getMap("documents").put(document.getDocumentId(), document);
        logger.info("Document added to cache");
    }
```

I've also enabled Hystrix Dashboard with Turbine using AMQP.  When I run a single instance of the micro service I see the hystrix commands displayed on the dashboard.  However when I start up another instance of the micro service (same host or even on a different host) I see that the dashboard stats seem to flip values continuously.  For example if I've invoked a command on the dashboard I'll see the success count flip between 0 and 1 continuously and the thread pool stats do the same for example the core size will flip between different values continuously.  Not sure if I've explained this adequately but i can try to record a video of what happens if anyone needs more clarification.  

Is this something anyone else has seen? 

Thanks

**Comment by aminmc** [comment]
Moving this question to stack overflow. Not sure if this warrants being an issue in Github.

**Comment by aminmc** [comment]
Hi 
Apologies for reopening this issue but I've noticed that if you use Turbine AMQP and Hystrix and have multiple instances of the same service deployed then you get a single view of the services on the dashboard.  I'm not sure if this expected behaviour but I should be able to view the different instances of the micro service on the dashboard and see their stats separately.  I noticed this line in HystrixStreamTask 

```java
if (this.properties.isPrefixMetricName()) {
  name = localService.getServiceId() + "." + name;
}
```
If i change this line to something like:

```java
 if (this.properties.isPrefixMetricName()) {
          String prefix = localService.getServiceId() + "[" + localService.getHost() + ":" + localService.getPort() + "]";
         name = prefix + "." + name;
 }
```
Then I see the the different instances on the dashboard: 
<img width="1280" alt="screen shot 2015-08-02 at 19 07 23" src="https://cloud.githubusercontent.com/assets/290639/9027341/c21c55c6-3949-11e5-969d-e14d4c746d5d.png">

Any help on this would be appreciated.

Thanks
